File: game.js
function startGame(){
  const numberToGuess=Math.floor(Math.random()*100)+1;
  let attempts=0;
  let playerGuess=0;

  function getGuess(){
    const playerGuessInput=prompt)"Enter your guess(between 1 and 100):");

    if(playerGuessInput===null){
      alert("Game cancelled.");
      return;
    }

    playerGuess=parseInt(playerGuessInput);

    if(isNull(playerGuess)){
      alert("Please enter a valid number.");
      getGuess();
    }else{
      attempts++;
      checkGuess();
    }
  }

function checkGuess(){
  if(playerGuess < numberToGuess){
    alert("Too low! Try again.");
    getGuess();
  }else if(playerGuess > numberToGuess){
    alert("Too high! Try again.");
    getGuess();
  }else{
    alert(`Congratulations! You've guessed the number in ${attempts} attempts.`);
    askReplay();
  }
}


function askReplay(){
  const playAgain=confirm("Do you want to play again?");
  if(playAgain){
    startGame();
  }else{
    alert("Thanks for playing! GoodBye.");
  }
}
  getGuess();
}
startGame();




  

File: game.java
import java.util.Random;
import java.util.Scanner;

public class NumberGuessingGame{
  public static void main(String[] args){
    startGame();
  }

  public static void startGame(){
    Random random=new Random();
    int numberToGuess=random.nextInt(100)+1;
    int attempts=0;
    int playerGuess=0;
    Scanner sc=new Scanner(System.in);

    System.out.println("Welcome to the Number Guessing Game!");
    System.out.println("I have selected a number between 1 and 100.Try to guess it.");

    While(playerGuess!=numberToGuess)
    {
      System.out.print("Enter your guess:");
      try{
        playerGuess=Integer.parseInt(sc.nextLine());
        attempts++;

        if(playerGuess<numberToGuess){
          System.out.println("Too low! Try again.");
        }else if(playerGuess>numberToGuess){
          System.out.println("Too high! Try again.");
        }else{
          System.out.println("Congratulations! You've guessed the number in "+ attempts + " attempts.");
        }
      }
      catch(NumberFormatException e){
        System.out.println("Please enter a valid integer.");
      }
    }

    System.out.println("Do you want to play again?(yes/no): ");
    String playAgain=sc.nextLine().toLowerCase();
    if(playAgain.equal("yes")){
      startGame();
    }
    else{
      System.out.println("Thanks for playing! Goodbye.");
    }
  }
}
          
  
